TASK-52261: In User Profile, no gamification points and no notification added when changing profile banner.

IlhemEssaadi opened the pull request:
ISSUES : no gamification points and no notification added when changing profile banner.
FIX : the problem is that the profile update does not support editing banner, so this has been fixed by adding the banner in the list of detected changes.

IlhemEssaadi committed:
In User Profile, no gamification points and no notification added when changing profile banner.

ISSUES : no gamification points and no notification added when changing profile banner.
FIX : the problem is that the profile update does not support editing banner, so this has been fixed by adding the banner in the list of detected changes.

ghost commented:
### CodeSee Review Map:
[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/Meeds-io/social/1190/5e2900d5/a405e02777295b5ba70a75d0aa6f31b3d97e483e.svg)](https://app.codesee.io/r/reviews?pr=1190&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

Review [in an interactive map](https://app.codesee.io/r/reviews?pr=1190&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

View [more CodeSee Maps](https://app.codesee.io/r/maps?src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

ahamdi reviewed (dismissed):
Please add unit tests for this change in org.exoplatform.social.core.manager.IdentityManagerTest#testUpdateProfileAndDetectChanges

IlhemEssaadi committed:
TASK-52261 : In User Profile, no gamification points and no notification added when changing profile banner.

ISSUES : no gamification points and no notification added when changing profile banner.
FIX : the problem is that the profile update does not support editing banner, so this has been fixed by adding the banner in the list of detected changes.

IlhemEssaadi commented:
I added unit tests for this change.

ahamdi commented on component/core/src/test/java/org/exoplatform/social/core/manager/IdentityManagerTest.java:
```diff
@@ -589,6 +593,15 @@ public void aboutMeUpdated(ProfileLifeCycleEvent event) {
+    BannerAttachment bannerAttachment = new BannerAttachment(null, "banner", "png", inputStream, System.currentTimeMillis());
+    profile.setProperty(Profile.BANNER, bannerAttachment);
+    identityManager.updateProfile(profile, true);
+    assertEquals(5, changes.size());
```
Why is there 5 changes instead of 4 ? the previous test asserts that there were 3 
Wht is the extra property changed ?

rdenarie replied:
resolved by creating a test dedicated to update banner

rdenarie committed:
Merge branch 'feature/maintenance' into TASK-52261

rdenarie reviewed (approved):
LGTM 
Kudos for the 100% test coverage !
